File: Algorithms & CP/First_problem.cpp
#include <bits/stdc++.h>
 
using namespace std;


int main() {

	vector < vector<int> > Children;
	vector < vector<int> > Boxes;
	
	// dealing with inputs
	int n; cin>>n;
    for (int i=0; i<n; i++){
    	string boxId;
		int boxX, boxY;
    	cin>>boxId>>boxX>>boxY;
    	Boxes.insert(Boxes.end(), { {boxX, boxY} } );
	}
	int k; cin>>k;
	for (int i=0; i<k; i++){
		string childId;
		int childX, childY;
    	cin>>childId>>childX>>childY;
    	Children.insert(Children.end(), { {childX, childY} });
	}

	// choosing the smallest distance for each child
	for(int i=0; i<k; i++){
		int boxNear=0;
		float distance, min = 99999999999;
	    for(int j=0; j<n; j++){
	        distance = sqrt( pow( (Boxes[j][0]-Children[i][0]), 2) + pow( (Boxes[j][1]-Children[i][1]), 2) );
	        //cout<<j<<endl<<distance<<" < "<<min<<endl;
	        if (distance < min) {
	        	min = distance;
	        	boxNear = j+1;
			} 
			//cout<<distance<<endl<<min<<endl;
		}
		cout<<"p"<<(char)(i+65)<<" Box"<<boxNear<<endl;
	}
	
    return 0;

}

File: Algorithms & CP/Challenges/first_problem.py
import math
 
B = int(input())
boxes = []
Children = []
for i in range(B):
    boxes.append(tuple(input().split()))
K = int(input()) 
 
for i in range(K):
    Children.append(tuple(input().split()))
 
def calcDist(child, boxes):
    distances = []
    childId, childX, childY = child[0], int(child[1]), int(child[2])
    for box in boxes:
        boxId, boxX, boxY = box[0], int(box[1]), int(box[2])
        distance = math.sqrt((boxX-childX)**2+(boxY-childY)**2)
        distances.append((boxId, distance))
    return distances
 
for child in Children:
    distances = calcDist(child, boxes)
    boxNear, distNear = min(distances, key=lambda x:x[1])
    print(child[0], boxNear)
 